---
title: "Final MBiol Analysis"
output: html_document
date: "2024-04-02"
---


## 1. Prepare data for downstream analyses- load packages, raw counts, TPM values, and metadata

```{r include=FALSE}
#load packages 
library(GenomicFeatures)
library(DESeq2)
library(Rsamtools)
library(GenomicAlignments)
library(GOstats)
library(GO.db)
library(Category)
library(Rsubread)
library(dplyr)
library(edgeR)
library(ggplot2)
library(plotly)
library(manhattanly)
library(ggvenn)
library(knitr)
library(tidyverse)
library(ggrepel)
library(htmlwidgets)
library(topGO)
library(devtools)
library(ggfortify)
library(tidyr)
library(ggpubr)
library(factoextra)
library(tidyr)
library(tidyverse)
library(dplyr)
library(ggplot2)
library(reshape2)
library(ggpubr)
library(tibble)
library(scales)
library(viridis)
library(rrvgo)
library(pheatmap)

```


```{r}
#load raw counts of all libraries used in the analysis
counts<- read.csv("/drives/raid/AboobakerLab/kriszta/ACME_vs_live/counts_tpm_matrices/counts_all_datasets.csv")
rownames(counts)<-counts$X
counts<- counts[, -1]

#make a vector of transcript lengths
lengths <- exonCounts_pu$annotation

# filter for transcripts that are between 500 and 5000 basepairs to avoid annotation issues 
matching_indices <- match(rownames(counts), lengths$GeneID)
counts$lengths <- lengths$Length[matching_indices]
counts<- counts[(counts$lengths> 500 & counts$lengths<5000), ]
counts<- counts[rownames(counts) %in% rownames(ratio_high), ]
counts$gene<- rownames(counts)
#remove the libraries not used in the analysis
counts<- counts[, -c(3:4)]
counts<- counts[, -c(7:8)]


#load metadata 
metadata<- read.csv("/drives/raid/AboobakerLab/kriszta/ACME_vs_live/annotations_functional/matrix_all_datasets.csv")
rownames(metadata)<- metadata$library

#remove metadata related to libraries not used in the analysis 
metadata<- metadata[-c(3:4), ]
metadata<- metadata[-c(7:8), ]


#load tpm values
DE_matrix<- read.csv("/drives/raid/AboobakerLab/kriszta/ACME_vs_live/counts_tpm_matrices/tpm_all_datasets")
rownames(DE_matrix)<- DE_matrix$X
DE_matrix<-DE_matrix[, -1]
DE_matrix_all<- DE_matrix
DE_matrix<- DE_matrix[rownames(DE_matrix) %in% rownames(ratio_high), ]
#remove TPM values not used in the analysis and prepare pca matrix
DE_matrix<- DE_matrix[, -c(3:4)]
DE_matrix_pca<-DE_matrix[, -c(7:8)]
DE_matrix<- DE_matrix[, -c(7:8)]


#load list of transcription factors in the Smed genome 
tf<- read.csv("/drives/raid/AboobakerLab/kriszta/ACME_vs_live/annotations_functional/tf.csv")
rownames(tf)<- tf$X
tf<-tf[, -1]

#load table containing correspondence between different annotations of the Smed genome
gene_correspondence<- read.csv("/drives/raid/AboobakerLab/kriszta/ACME_vs_live/annotations_functional/gene_correspondence.csv")
rownames(gene_correspondence)<- gene_correspondence$X
gene_correspondence<- gene_correspondence[, -1]
gene_correspondence$transcripts <-(sub("\\.[^.]*$", "", gene_correspondence$V5))



```


```{r}
#load pfam data 

pfam<- read.table("/drives/ssd1/simon/smed/sexual_asexual_comparison_to_gtig/pfam/longest_orfs_formatted.pep.tsv", fill = TRUE, sep = "\t", header = F,
                          quote = "", stringsAsFactors = FALSE)  
 # only keep columns of interest
pfam_small <- pfam %>%
dplyr::select(V1,V5,V6,V12,V13,V14)

#reformat names
gene_names <- as.character(pfam_small$V1)
gene_names<- gsub("\\.\\d+\\.\\w+$", "", gene_names)
pfam_small$V1 <-gene_names



```


```{r}
# make GO annotation table for gene universe- define gene universe as all genes that have a GO annotation
GO_universe_format<- pfam_small %>%
  filter(grepl("G", V14)) %>% #  isolate all the rows that have a GO annotation
  group_by(V1) %>%
  summarise(GO_terms = toString(V14)) %>%
  ungroup()

# remove the | sign and replace with comma 
GO_universe_format$GO_terms<- str_replace_all(GO_universe_format$GO_terms, "\\|", ", ") 
#save
write.table(GO_universe_format, "/drives/raid/AboobakerLab/kriszta/ACME_vs_live/annotations_functional/GO_universe_format.tsv", sep = "\t", col.names = F, row.names = F, quote = FALSE) 

#prepare GO table
geneID2GO = readMappings("/drives/raid/AboobakerLab/kriszta/ACME_vs_live/annotations_functional/GO_universe_format.tsv") 
gene_all_names = names(geneID2GO)
geneNames <- names(geneID2GO)

gene_names <- setNames(gene_correspondence$V4, gene_correspondence$transcripts)
names(geneID2GO) <- gene_names[names(geneID2GO)]
filtered_geneID2GO<- geneID2GO[names(geneID2GO) %in% rownames(counts) ]
counts
```


##2. Calculate gene body coverage 

```{r}
#gene body coverage data calculated using geneBody_coverage.py
#Aboobaker_G2_3
W1_sorted <- c(0.0,0.0030237226195592424,0.005947615182019092,0.009141327062893091,0.01273505458277913,0.016422353464234336,0.02059018108028794,0.025217984183890055,0.029093971145404643,0.03338859529682984,0.03996370265663854,0.04575562312886252,0.050449876580069504,0.05731682459441721,0.06436867456436775,0.0717944236413647,0.0793675225007831,0.08764661766958012,0.09813031902193742,0.10707360859539883,0.11631839730415576,0.12512430464733643,0.13588895188793937,0.1482478665961366,0.15958120518368343,0.1712101705502259,0.18362417723265828,0.19896988053000178,0.21354360077743817,0.2292414136492414,0.24468285180784352,0.2619383844560583,0.27861595361580294,0.295394898189568,0.31375644278617343,0.3317530513458745,0.3506160642633585,0.3687664358367034,0.3869273931049574,0.4061258257421538,0.4257821476350553,0.4443702415562857,0.46227443777210836,0.4829343917173814,0.5006894609903234,0.5188946154665193,0.5366489893288469,0.5542516091415279,0.5733781826818959,0.5897655338118686,0.6092523300505316,0.6261450128936855,0.6435091068656036,0.659524258781143,0.6777274042933416,0.6990401170020633,0.7132385473212243,0.7331148507718362,0.752216080458699,0.7702446779066657,0.7898126053836836,0.8069952744530712,0.8258999347395772,0.8418604582879577,0.8574982340433784,0.875954740822782,0.8910920527393229,0.9064364424832835,0.9212731052108354,0.9373511531530613,0.9502370345713342,0.9597980033061366,0.9716599312285263,0.9838442206046456,0.9891886829803815,0.9985968931835388,1.0,0.9991633437629472,0.9981347541963008,0.9926485825909053,0.9829510815721287,0.9702174950784246,0.9540621020223005,0.9334987328845926,0.9065840240692431,0.8751499962061488,0.8407284070754476,0.7998616751019975,0.7569583945101505,0.7052587105201872,0.6485855270833846,0.5876352592962104,0.5230832281331609,0.45302578567104884,0.37938520138395987,0.30698940209677117,0.23480160785115492,0.16815994505947074,0.11081553584401291,0.06896224801216876)

#Aboobaker_G2_4
W2_sorted <- c(0.0,0.0025833501759512357,0.005782321849698729,0.008620324614832276,0.01285433793963779,0.016275229505564818,0.020690571580433328,0.02525944559124343,0.030266898774394303,0.034947024987517415,0.04209944401176091,0.047468385670144816,0.05203167433979969,0.05889664313203071,0.0662605907160719,0.07300398091943548,0.08113123681274975,0.08929629117760293,0.09979283579980266,0.10857520025396417,0.11821718767941285,0.127117298918626,0.13846528333786223,0.1506093737092017,0.16221636208805512,0.17462776948491723,0.18650285423922197,0.20149611805800546,0.21569658816188242,0.23167352148226122,0.24677732309925646,0.2637406536459808,0.2792941401717999,0.2953324472990589,0.3129096458062401,0.3299237639899807,0.348557696054567,0.36682085239060397,0.3852536472277744,0.4043996121954755,0.4226565337320834,0.4421863934854659,0.46045312184200915,0.4810993354223375,0.49894242214683887,0.5173602794437104,0.536065846755904,0.5542230764474736,0.5739577103946056,0.591313768359536,0.612215868499767,0.6293565011552564,0.6476895392015612,0.664196690256723,0.6829921425940195,0.7031442483195917,0.7171656627773642,0.7373146511032218,0.7571939194079434,0.7756208691206491,0.7945303663308361,0.8112595671698323,0.8305205511978349,0.8466835543136759,0.86133039708918,0.8775798431012723,0.8911045518711932,0.9046258185122625,0.9176812936793162,0.9319313823953164,0.9431553202842445,0.953891969630292,0.9662507059611437,0.979443411661465,0.9863832629926725,0.9966975696232259,0.9986331501168245,1.0,0.9992357824491436,0.9956108960393956,0.9880139877725191,0.977503414851604,0.9626722558179771,0.9421989929759789,0.9174958733421279,0.887665475473598,0.8545896047189119,0.8137814136472485,0.7674967955728768,0.7145836115179607,0.6583812460350572,0.5960850785143097,0.5292766047984055,0.45868458201644846,0.3829236507244577,0.30922409999371325,0.23681833509033445,0.17034238401064905,0.11249474588256443,0.06988352875099822)

#Aboobaker_G1_2
G1_2 <- c(0.0,0.001328743385765659,0.01665323980473682,0.015002399145594602,0.027890343936500403,0.03348945592095128,0.04337165087014173,0.05254331544968686,0.06409182941669436,0.08197486167138265,0.10240647938835423,0.11199661245064431,0.09676738164399333,0.10710545110939293,0.12193261312284731,0.1199804972680697,0.13140500009950373,0.13184189519971873,0.14403626210905168,0.14719495287576848,0.15321474453706072,0.16840288398675346,0.17078397149623398,0.18420205007175325,0.19162060626519736,0.20659168016109286,0.2070429480229345,0.21592992136256728,0.22509624222304608,0.24276018199601543,0.2476861696442852,0.25760171677109645,0.2683295091959877,0.2818704933100323,0.29599467986700406,0.30852699094073777,0.3251175802460764,0.33495960516918216,0.34889934128527866,0.363508700680237,0.37952861764252804,0.4036380039403479,0.4139722038160788,0.4351316729231175,0.44779739270733443,0.451494693502701,0.47530851685630116,0.4812046659143693,0.4969714011526218,0.5082169815284215,0.5206975138070644,0.5341229630295717,0.5516673140551605,0.5687906168715584,0.5855487041296998,0.6000721954872477,0.6140285155590672,0.6350261989647189,0.6597033093467911,0.6773740669682244,0.695485220747192,0.7119501405213848,0.7253448172927172,0.7448621062508246,0.7634011260874468,0.7849705874331943,0.8071901398432854,0.8217902859295086,0.8382144828790926,0.8568782509159871,0.8759357957682168,0.8891865606571079,0.9063597996068496,0.9259150473969454,0.9421068847002321,0.9610306523096291,0.9695583066085958,0.9786369167699172,0.9853290957026917,0.9881292044934412,0.9929009613534866,0.9982785853959475,1.0,0.9928970917638179,0.9817600441649168,0.9653332674888869,0.9523203902315341,0.9336216116214465,0.9070546673516454,0.8697310008675252,0.8334852915054031,0.7895160654307074,0.7331234295915261,0.6718908584076602,0.5967242633591134,0.5153663248406282,0.42316413767484096,0.3255225972980682,0.2235520916790555,0.13010739769726246)

#Aboobaker_G1_2
G1_1 <- c(0.0,0.002593814024157098,0.01692883926223736,0.015032530837362735,0.02748737217918534,0.03300994340609753,0.04265464660221858,0.05139206567823838,0.06385425086477763,0.08256684614916722,0.10269377013670247,0.11348107978145357,0.09939686280532634,0.11152075218850564,0.1253546398490744,0.12361893811517671,0.13430263568990392,0.13504548154787174,0.14781355383173878,0.15255783716748547,0.1588622085591233,0.17300916751293835,0.1750852404845724,0.18730459984906803,0.19391517763563004,0.20803467699615677,0.20787279311479434,0.21706026215290555,0.22603699477740888,0.24240179483564875,0.24821716195843735,0.258510198923999,0.2697512304531998,0.282293240038833,0.296338183410606,0.30929607811551835,0.3263045821280514,0.3353525181165465,0.3496308680312701,0.36511201199154053,0.3802851934201067,0.4037136545541656,0.4145852587643598,0.43658454385704154,0.45013904612078165,0.4551306254449212,0.47726848569443,0.48285794976778285,0.49532380687668043,0.5083699873669906,0.5209092829230545,0.5348632264798605,0.5540180489352303,0.5705088118952404,0.5868950048924374,0.6015368751206346,0.6154960870877808,0.6364787286336027,0.6605207200990972,0.6766980924682646,0.6933129025444665,0.7088258166155764,0.7230326438687118,0.7422596275808625,0.7611612470423065,0.7815479196244932,0.8032569632020695,0.8173711941522561,0.8336657487392933,0.8520312675364227,0.8715836154992801,0.8829043117787766,0.8984017399164622,0.9184778955861738,0.9353382455139885,0.9529877394512456,0.9622301276759334,0.9736148431235032,0.9825355390198426,0.9856319274606948,0.992098980977686,0.9975653558276362,1.0,0.9940024735984792,0.9832469340645674,0.9684554729124683,0.9562094521986753,0.9386260528638679,0.9124675972807978,0.8769537221239229,0.8393047039400845,0.7957512346040686,0.7401527423911782,0.6796685180073467,0.6040893401060611,0.5219275229578964,0.42960884926902404,0.3286948439345191,0.2270126085831345,0.1307858919634232)


#HEX2 <- c(0.0,0.015018401929587106,0.032085268937002503,0.049758014646971396,0.07200782318728137,0.09508264013801426,0.11791892680562753,0.14246995498982373,0.167158638882574,0.19100421838364845,0.21500180412724723,0.23719804014631315,0.2603356459436259,0.28400939779015244,0.3050741299767841,0.32602818209203177,0.3476344049548985,0.3677959372586238,0.3880226571785369,0.40604191988766014,0.42258787513705603,0.4377386723155306,0.45441891657929395,0.47100304172403024,0.48407216980326134,0.4956670645818637,0.5092989355943827,0.5248916113848467,0.5358658140064747,0.5505836543125543,0.5645506797313479,0.5786907111476547,0.592228903894432,0.6056662220666436,0.619873334864618,0.6341255613646929,0.6492712243345329,0.6599821935544471,0.6726631838776422,0.6849280507993752,0.6960563638473941,0.7069926490757457,0.7194659564513057,0.7325994304731779,0.7428205456889438,0.7523841034463923,0.7624502665622206,0.7729173187883167,0.783762029934793,0.7928635089151325,0.8049918143039221,0.8141671501379798,0.826113696124371,0.8331637220857916,0.8438331126325015,0.8519122949230841,0.8572366376118152,0.8672240200836775,0.8782055452651613,0.8897763681803296,0.9001868178604137,0.9100512739107881,0.9196385247785747,0.9285361504259163,0.9367754611087483,0.9472974325337631,0.9556917794838875,0.9606701101775923,0.9671813388181271,0.9715264364190124,0.9746930648035605,0.9793747476557498,0.9887372296029043,0.9966933171372898,0.998382640112764,1.0,0.9976964236701832,0.9948647813572002,0.9900419919349157,0.984281830683731,0.9765008533307252,0.9642921260345536,0.9477815631577942,0.9287163527322533,0.9074225170062244,0.8827675421390173,0.8566827322507672,0.8239973921586831,0.7820362204110717,0.7320060085391148,0.6784364331524774,0.6191614473788645,0.5542548239046398,0.48745039491743647,0.41353138360778485,0.3384616362251581,0.2614240560904718,0.18555564467943222,0.11248861741737354,0.05989407875084872)

#HEX1 <- c(0.0,0.014554576258058963,0.03192680889643046,0.04998973615618833,0.07175617842418772,0.09537610361483329,0.11777697268897323,0.14275532947034492,0.1682446018484121,0.19204411435057905,0.21644004652303483,0.23900056473906903,0.26261224626342644,0.28741806119225555,0.30917937513297106,0.33017896472300345,0.352002489587219,0.37246187939391606,0.39428569182788586,0.4127563929991527,0.4301319326896979,0.44633457120929265,0.464018433604673,0.4812181243905019,0.4947770383028055,0.5066770102312046,0.5200019382201435,0.5362326627190689,0.5461528606744191,0.5608773427289402,0.5745765906813117,0.5885036897356242,0.6012744228798549,0.6149494670445781,0.630433852174672,0.6453590579176508,0.6611803789431265,0.6727118302314702,0.6849993024037045,0.6973147646986847,0.7081424369418241,0.7192307432388876,0.7307552449248371,0.7434797272669279,0.7543098917812708,0.7639508595052563,0.7752391226380998,0.7868500869634901,0.7974519690212607,0.8059051298753678,0.8174064342678088,0.8272157261546081,0.8390287082323504,0.8454631773400019,0.856091228245408,0.8643835417503115,0.8696664857054107,0.8787589141231881,0.8901857379492816,0.9021122621516554,0.9120947668870905,0.9220953405887502,0.9312402504866759,0.9398592911606857,0.9481589376943222,0.9583585575942067,0.9666144455969069,0.9709261227070686,0.975628319543485,0.9790528918901159,0.9800036454259179,0.9841826570227913,0.9931977291766787,0.9983917182211337,0.9984776536493585,1.0,0.9972720653830096,0.9927626840668096,0.9886393651456735,0.9827417887691837,0.9734779208495884,0.9604765567624591,0.9449645649359585,0.9257853398185425,0.9030769628205378,0.8766339174242492,0.8500968846466173,0.8187864815183952,0.7778295509842722,0.7283058695431183,0.6762203729463385,0.6183463368382175,0.5564572038931385,0.48909207183128206,0.4141050472182347,0.3395035577890353,0.2619597504910613,0.18542071191439355,0.11063939553221086,0.058244186310117234)

#Pearson_X1_1
s1r1_sorted <- c(0.0009672435050870168,0.047524716009239236,0.09849441489221768,0.1364878800793438,0.18333079026337434,0.2283400593149145,0.26855081843006645,0.2958997819975201,0.32744612637830006,0.3661691895674501,0.39624581489607613,0.42490445949240324,0.4563398734077313,0.4868913512821457,0.5136847856016641,0.5405268890186732,0.5589045156153265,0.5788290302835786,0.6013514224530278,0.6341035327626333,0.6498100589816846,0.661000005261524,0.6778566129465061,0.6943317601376414,0.7029584672829665,0.7084940290471204,0.7196089986092038,0.7350717408804635,0.7440702623922045,0.7626329192513903,0.7749387470908157,0.7832010936954665,0.7950259305443749,0.8035417072242479,0.8114002318578266,0.8351139207649555,0.8535191703628873,0.8620055702001308,0.8665015425034682,0.8750309114538117,0.8752247109230993,0.8913648744161901,0.9044309924812821,0.9183749081425593,0.9254884886622926,0.9275992367282441,0.934686509627712,0.9409792923951685,0.94706994494692,0.9530518593349083,0.9575026702234569,0.9650384880484482,0.964963511330692,0.9630014013192395,0.9666476374880081,0.9679397801033714,0.9659180394859842,0.970459611664448,0.9735858338726395,0.9808695370034217,0.9877673950369797,0.9936905557397089,0.9983110507789686,0.99758627584066,1.0,0.9960170262918356,0.9963178100835355,0.9986543652234307,0.9962384487624019,0.9889012534704136,0.9850493794032379,0.9922107520997866,0.9936094405772242,0.9929136040212074,0.9930354959951033,0.98471308032418,0.9713510015311035,0.9623244185577461,0.9403286172539405,0.9306741590769182,0.915614800316393,0.9060410188415177,0.8841570249238394,0.8639540879411131,0.8440769269893384,0.8085037628666186,0.7872849132813144,0.7594505741199663,0.7347823570575452,0.6827384303470677,0.6358963970837126,0.5838331781183738,0.5157407261253961,0.45783677700082603,0.3863800188362561,0.31265510534448077,0.23267117053125608,0.16044622985493978,0.07942665172393835,0.0)

#Sanchez_X1_2
s3r2_sorted <- c(0.0,0.013712869682636313,0.028206330716914,0.048017497117200776,0.07493989368494178,0.10912737384312184,0.13650664478876198,0.17078198540423034,0.21514956388021456,0.2404400988879209,0.27760351035718595,0.30975614232456905,0.3486732485213085,0.38850488445568887,0.43189428317386336,0.4661571280798507,0.48057639583908496,0.5112658583244268,0.5386970645625975,0.567337230693055,0.5850323267909183,0.5999342413288562,0.6205287325172334,0.6520757130657091,0.6713452685581787,0.6806241138296655,0.698184881050608,0.7169495317818608,0.73481097701219,0.7570728644344383,0.771173491601907,0.7834653648220006,0.7893094519498969,0.8013857741814432,0.8148456057470892,0.839584377083025,0.8647011436307611,0.8631079406719333,0.8786916523584285,0.8941422066407667,0.8916782089274815,0.8983122591891299,0.9160838915075644,0.9256556049700122,0.9250729925154605,0.9393286445786778,0.9541020876035142,0.9629174201514402,0.9660745392499998,0.9876366669413119,0.9912147352530127,1.0,0.9955038875323668,0.9837922838212972,0.9799221283004781,0.9722599154430959,0.9788955276684295,0.9649542207968436,0.9712797832306798,0.978961520634126,0.9878166832560227,0.9832877695510018,0.9730393353219734,0.9752491234455044,0.9690196217783035,0.974814116559197,0.9913810843854786,0.9881255615747562,0.9820389795847442,0.9714457418722243,0.9747637432303518,0.9765420388858669,0.9738004021275507,0.9768923092422562,0.9648718272124532,0.9367611667712532,0.9259316820513581,0.9118743993761517,0.8851027323040253,0.8818175321832855,0.886451487946131,0.872748771027448,0.8589671749426662,0.8520988301282489,0.8248413923500486,0.7881692184597554,0.7704381971971342,0.7446923497752139,0.702952775199453,0.6601131876984426,0.600461404072586,0.5459773772989664,0.49514092622103584,0.46148099929750686,0.37965714116129656,0.2952173843434227,0.22045711648227034,0.16275427304452886,0.08427848406738468,0.04407822470331476)

#Reddien_X1_2
s2r2_sorted <- c(0.0555715465592681,0.09807809023888527,0.15099964661716597,0.19376074788103675,0.24679789677512123,0.2977818177291482,0.3435694105111958,0.3779433220679323,0.4162316280939099,0.4590481121455724,0.4898444988505531,0.5247305741696202,0.5510846515749417,0.5817773860899269,0.6137711080190466,0.638155793814454,0.6559130906872827,0.6790877387842869,0.7027801397322979,0.7326817641501118,0.7491314683221886,0.7646175627965869,0.7792881584232266,0.7970764493332598,0.8080158201695475,0.8132743904654212,0.8242958192854554,0.8379400549915516,0.84392546086482,0.858904980665564,0.864441227048852,0.8755594487227789,0.8816391069550874,0.8906621825238851,0.8998214286167946,0.9144811001155672,0.9302204819267926,0.9347969293546813,0.9399307613532809,0.9440913297737613,0.9444015241952919,0.9547266033761652,0.9678995772870614,0.9738666915973895,0.9759440542385489,0.9818082785073373,0.9865353772799988,0.9873554490184925,0.9909512654331887,0.9921933133667374,0.9938977313635013,0.9997337561394234,1.0,0.9966501034870578,0.9889125183137922,0.984002503911727,0.9786806752940178,0.9718807867709319,0.9766109341374168,0.9834822105658673,0.9878404294858978,0.9839768449137134,0.9825988805055381,0.978078578013897,0.9688624248362842,0.966849082665416,0.9628193497298565,0.9569274340671722,0.9459883172803698,0.9357727334276534,0.9271881472704542,0.926646767817316,0.9205475477746409,0.9119898908628816,0.9054745377633128,0.8902590059907409,0.8717921489055167,0.8514638712416237,0.8304194280787177,0.816964205189672,0.7980471724165263,0.778107336415631,0.7532333513115685,0.7346945982220092,0.7044677904630331,0.6739287558866441,0.6460709595779324,0.6064374615273811,0.5819478532945518,0.5300668836079905,0.48642804137246676,0.4323167712546056,0.38044291495363214,0.3289976401343312,0.2700248790660938,0.21038549214847363,0.151634262231403,0.10645994490682863,0.04751233473763421,0.0)

#Reddien_X1_1
s2r1_sorted <- c(0.05483966215218692,0.09655547526360365,0.1508167763558332,0.19338731024551206,0.24552113361875827,0.29714730465097605,0.3428259140979717,0.3757127205762372,0.41333882426681795,0.4569733560222849,0.48808001222509717,0.5204563172930734,0.5472823764345711,0.5780170995956911,0.6093340689112152,0.6345094809254813,0.6519646922616689,0.6741972747357489,0.6992701202566234,0.7300897974829769,0.7451846844396788,0.7604973956916893,0.7761478511542615,0.7943603969012487,0.8053971732052806,0.8090100520343649,0.8205226229049628,0.8333605289944599,0.8387261034315744,0.854061607237672,0.8604979137042822,0.8721298864775403,0.8789228446143525,0.8871012274308389,0.8962545099471368,0.9124890246081883,0.9288507113607932,0.93387387947401,0.9387911140119817,0.9442077126894954,0.9431706514785374,0.9528613720599548,0.9648699399882411,0.9710661476180485,0.9734005713678899,0.980108057426876,0.9846764104835388,0.9846929868865112,0.9870846510278665,0.9907594323618008,0.991372241259185,0.9972428779743635,1.0,0.9964075826683346,0.9895066189059056,0.9855298362303188,0.97948955039097,0.9722026672468408,0.9749245644161609,0.979375587620535,0.9830889608926393,0.9789080812554554,0.9777353007451611,0.9733021489752416,0.9640134061659039,0.9635679153360218,0.9604727900935272,0.9549300553496456,0.9431180732003595,0.9314117656200223,0.9238143339264577,0.9232706797102238,0.918069833277647,0.909689943562528,0.904624039410398,0.8885599508924062,0.8688368804245631,0.8469290918462228,0.8259467327650735,0.8120746145338793,0.7916789047141736,0.7724080592399202,0.748943124807364,0.7302104944171193,0.7005814691355147,0.6712293215848077,0.6435744422946922,0.6057351764221388,0.5822562556495748,0.5293011233103077,0.4849929161777923,0.430573362238436,0.38056313148972654,0.32948683082485736,0.2700635342445175,0.20890230541504465,0.15075409683209398,0.1067297606004802,0.047246892571958424,0.0)
#Sanchez_X1_3
s3r3_sorted <- c(0.0,0.01497939166507343,0.03188077278931802,0.052814491840942536,0.08080844010042042,0.11838565819953847,0.1499697453394282,0.1852223869128484,0.23020259911180238,0.2576359165868701,0.2963416719246855,0.33134532390675986,0.371256011016981,0.41434131541686153,0.45829646657092427,0.4920410234603975,0.506880964786162,0.5377285924529477,0.5604836541971117,0.5896409012291436,0.6064497196643976,0.6197080111996812,0.6383345344925362,0.6713159544526898,0.6921672159123751,0.7042771642187405,0.7237925318482343,0.7396809780440493,0.7536086312080158,0.7770051645130011,0.7910007239291528,0.8032242535263757,0.8066139072812883,0.818114316133515,0.8289465759687049,0.8505088320569248,0.874479334874428,0.8731571839762263,0.8862517729392777,0.9001955942585264,0.8980892878291382,0.9078803184641546,0.9252740199167421,0.9334138776935641,0.9350994078817332,0.9473766965952695,0.9601968311224055,0.9655820394430232,0.9687655492180878,0.9879187501692603,0.9923209346487205,1.0,0.9931139826653124,0.9805537512343376,0.9760197962893276,0.9718581131196901,0.9763860050064693,0.9625553607741959,0.9660343435870427,0.974632164362233,0.9829005589735484,0.9755246465338102,0.9645978029902195,0.9691139729644191,0.9600003880357267,0.9641119499239894,0.9813160797558932,0.9719037881583622,0.961628925476526,0.9462380542595207,0.9463625490551937,0.9490816285697771,0.9479931075154032,0.9501155821000736,0.9410961766759001,0.9152032801953437,0.8997210589009939,0.8837408584239524,0.8609150610004288,0.859186685200762,0.8664934787766688,0.8507541838133322,0.8362367971854475,0.8294469803746889,0.8003176234105187,0.7646805839452643,0.7508285169072421,0.7260459684906906,0.6860394850604224,0.6413616173652456,0.5786425135984291,0.5231909956309603,0.4755030216261203,0.44204585369678806,0.3675943280494455,0.2848097351696585,0.20972401363137172,0.15755180175901445,0.0811600974778082,0.03896848786115112)
#Sanchez_X1_1
s3r1_sorted <- c(0.0,0.015900816814009112,0.0351848798153123,0.05851498986587088,0.08941647908655155,0.12743707394232032,0.15960893148176017,0.19689107972501768,0.24500177707477483,0.27164869312960477,0.30864346756918587,0.34516715709661205,0.3868603731536833,0.42934046511330054,0.473204594138535,0.5091679447729988,0.5215506338233363,0.5530773011517366,0.579107043940175,0.609988921299602,0.6243271930863786,0.6374739442414773,0.6566671522937072,0.6848005673831533,0.7041046424877766,0.7139710096666464,0.7320451409012171,0.7488328941343725,0.7660713199342962,0.788530903490431,0.7993126242751616,0.8104913851897627,0.8167563742551495,0.8285427026265485,0.8398843620621752,0.8629835324404199,0.8878409662163677,0.8901095382487328,0.9021760360666131,0.9142021094357867,0.9080840092087694,0.9187456573735795,0.9329674587190333,0.9410571513651457,0.9397587661017384,0.9523027527048359,0.9662764039691205,0.9725534002964993,0.9746862902683543,0.9922561164992587,0.9971130539750441,1.0,0.9927143936652888,0.9773130789501491,0.9716324433017088,0.9662115647543634,0.9707763255216755,0.9525613090797315,0.9562391334278972,0.9652617903307921,0.9741607724351687,0.9664449058790756,0.9559541610766191,0.9610844638837569,0.9556855986500635,0.9588883356654104,0.971332661993974,0.9620770642084333,0.9533625936966678,0.9423683443746778,0.9453889712498119,0.9463419476099145,0.939639894208017,0.9391704102641277,0.9329702604134981,0.9041432258229778,0.8921199541482688,0.8747902731572055,0.851029902885265,0.8493672973414321,0.8574269718325643,0.839430887800942,0.8257310021100762,0.809461562352911,0.7805856982302897,0.7503253967999847,0.725005683437343,0.7016259433705505,0.667631383460717,0.6288667386035074,0.5766087329616952,0.5224251627384242,0.4756148518624064,0.4426877375436664,0.36321167041826896,0.2781258104901845,0.20304400101181194,0.15453025990118824,0.07855190819407577,0.03594814143594046)
#Sanchez_X1_4
s3r4_sorted <- c(0.0,0.012268082557573914,0.025291299683634496,0.04299578029824554,0.07093086645167893,0.10558428537236655,0.1330936909159187,0.16917868514435833,0.2129222606902305,0.23597058237103286,0.2741044703395618,0.30906043237570147,0.34937858378018244,0.39283307163217934,0.4398622164830545,0.47685850971828014,0.4901467578753971,0.5213490650316834,0.5515687186728224,0.5804464998167208,0.59564908749816,0.6106783856592932,0.6361046472987426,0.6651752894852404,0.6841013592829973,0.6949480600340137,0.7101734811581285,0.7284418662588179,0.7440048961424938,0.7688607294551003,0.7829656809279185,0.7951946204409056,0.799267128038122,0.8132681057987566,0.828570182051854,0.8514154491888624,0.8728719537231355,0.8727463697884196,0.8834112184781265,0.89664319850865,0.8969803995281632,0.9055645507540541,0.9239059213232959,0.9344060430337001,0.9316664376526732,0.9441498069554743,0.9559734528609692,0.9658790895817118,0.9693942165336142,0.9909387929182077,0.9952070157383581,1.0,0.9940677900448637,0.9793516362404248,0.9737684517661872,0.9652475002477021,0.9706023502951834,0.9576349935556685,0.963817963640189,0.9671553159312599,0.9768921482722219,0.9739311400452509,0.9615758011174523,0.9662073203201738,0.9587146892021873,0.9617984271835397,0.9742430611818019,0.9655125312787585,0.955805871701332,0.9397531460202736,0.9409629107420339,0.9414689161413275,0.9385788546826702,0.9395354128347597,0.9243939861604873,0.8973171928076287,0.8825806520008008,0.8670857147051041,0.8417507868363572,0.8405222660723991,0.8433405652826474,0.8257885794459058,0.8123433513703939,0.8093134350753484,0.7789221228740943,0.736927752133194,0.7148001074802766,0.6872099694072642,0.6415279976742507,0.5988433230324606,0.543625942542088,0.4897214850055636,0.44426295481873307,0.41410527766485644,0.340662732518859,0.26434317359573306,0.19423471882042434,0.14735929196756184,0.07478319442309472,0.04007432285590915)
#Kriszta_R2_G2
R2_G2 <- c(0.0,0.009218176113067255,0.02045510343384639,0.03476539267957415,0.04977270847449414,0.06625798313219622,0.08430822561746824,0.10435883881583305,0.1260215646169272,0.1485364769867256,0.1728518423521686,0.19813325786986538,0.22362819744518286,0.25011269988340323,0.27641375543701596,0.3027329527775255,0.3291723394562262,0.35578156667936195,0.3839614907154035,0.41067980736764226,0.436896718449135,0.461852536633579,0.48859925950477456,0.5152981214776438,0.5404428768243088,0.5620723028981385,0.5851068819794408,0.6105790249672761,0.6312385815652949,0.6542933314491337,0.6760880626206742,0.6969018244073156,0.7173865275987305,0.7396011456538425,0.7578984506317221,0.7779171963491563,0.7947354682795034,0.8132377038369999,0.829253082343303,0.8459597584134625,0.8603820015809612,0.8727415713198401,0.885817860995003,0.9005846787596794,0.9099946565276358,0.9207276718465844,0.9294866936557574,0.9387243244481943,0.947687322006095,0.954849985922212,0.9621279456682127,0.96975549287356,0.9754133433081333,0.9779317097786738,0.9826274487384118,0.9873569648408589,0.9878103901594174,0.9915576239844925,0.9947004498566381,0.9974807980524314,1.0,0.999478065565134,0.9987272104245573,0.9955675556579877,0.9924783196695044,0.990626067098115,0.985363039236269,0.976823628531636,0.9706600757873147,0.96192755053532,0.951351724251523,0.9403965913969499,0.9332475338210052,0.9220467707201585,0.9079768601508131,0.8936990351553218,0.8750723135206322,0.8557318557961875,0.8355756144306934,0.8131074887744709,0.7901288866540541,0.7646588920357198,0.7371773074413432,0.7079564983820987,0.6758955448426182,0.6397933889180178,0.6029607873630997,0.5609854666383878,0.5173476659815357,0.47214991035928255,0.4294638373913447,0.38217345052131974,0.3354809071418605,0.28780978145472547,0.2394798224539703,0.1916532980396009,0.14491528083903854,0.1027189167634527,0.06656722898646969,0.036390634207031446)

x <- seq(0, 1, length.out = length(W1_sorted))

df <- data.frame(x = x, W1_sorted = W1_sorted, W2_sorted = W2_sorted, W1=W1, W2=W2)
```


```{r}
#plot coverage curves on one plot
coverage_plot<- ggplot(df, aes(x = x)) +
  geom_line(aes(y = s1r1_sorted, color = "Live-dissociated"), size = 0.8) +
  geom_line(aes(y = R2_G2, color = "Fixed- 1 day protocol"), size = 0.8) +
    geom_line(aes(y = W1_sorted, color = "Fixed- 2 day protocol"), size = 0.8) +
  xlab("Gene body percentile (5'->3')") +
  ylab("Relative coverage") +
  scale_color_manual(values = c("Live-dissociated" = "#dd5182",
                                 "Fixed- 1 day protocol" = "#444e86", 
                                "Fixed- 2 day protocol"="#ffa600"
                                 ),
                      labels = c("Live-dissociated", 
                                
                                 "Fixed- 1 day protocol", 
                                 "Fixed- 2 day protocol"),
                      breaks = c("Live-dissociated", 
                                 "Fixed- 1 day protocol", 
                                 "Fixed- 2 day protocol")) +
  labs(color = "Sample type:") +
  theme_classic()
coverage_plot

#save
ggsave("/drives/raid/AboobakerLab/kriszta/ACME_vs_live/annotations_functional/coverage_plot.png", width = 7, height = 4)
```
```{r}
#plot coverage curves next to each other 
plot3 <- ggplot(df, aes(x = x)) +
  geom_line(aes(y = s1r1_sorted),color= "#dd5182",size = 0.8) +
  xlab("Gene body percentile (5'->3')") +
  ylab("Relative coverage") +
  theme_classic() +
  ggtitle("Live-dissociated libraries")

plot2 <- ggplot(df, aes(x = x)) +
  geom_line(aes(y = R2_G2), color="#444e86", size = 0.8) +
  xlab("Gene body percentile (5'->3')") +
  ylab("Relative coverage") +
  theme_classic() +
  ggtitle("Fixed libraries- \nstreamlined protocol")

plot1 <- ggplot(df, aes(x = x)) +
  geom_line(aes(y = W1_sorted ), color= "#ffa600", size = 0.8) +
  xlab("Gene body percentile (5'->3')") +
  ylab("Relative coverage") +
  theme_classic() +
  ggtitle("Fixed libraries- \noriginal protocol")

# Arrange the plots next to each other
p<- grid.arrange(plot1, plot2, plot3, ncol = 3)
ggsave( "/drives/raid/AboobakerLab/kriszta/ACME_vs_live/coverage.png",plot=p,  height=3, width=9)
```


```{r}
# create plots with C50 values marked 

plot1 <- ggplot(data.frame(x = x, coverage = W1_sorted), aes(x = x)) +
  geom_line(aes(y = coverage), color= "#ffa600", size = 0.8) +
   geom_segment(aes(x = 0, y = 0.5, xend =  45, yend = 0.5),
               linetype = "dashed", 
               color = "gray")+
  geom_segment(aes(x = 45, y = 0, xend =  45, yend = 0.5),
               linetype = "dashed", 
               color = "gray")+
  xlab("Gene body percentile (5'->3')") +
  ylab("Relative coverage") +
  theme_classic() +
  ggtitle("Fixed libraries- \noriginal protocol")

plot2 <- ggplot(data.frame(x = x, coverage = R2_G2), aes(x = x)) +
  geom_line(aes(y = coverage), color="#444e86", size = 0.8) +
  geom_segment(aes(x = 0, y = 0.5, xend =  23.4, yend = 0.5),
               linetype = "dashed", 
               color = "gray")+
  geom_segment(aes(x = 23.4, y = 0, xend =  23.4, yend = 0.5),
               linetype = "dashed", 
               color = "gray")+
  xlab("Gene body percentile (5'->3')") +
  ylab("Relative coverage") +
  theme_classic() +
  ggtitle("Fixed libraries- \nstreamlined protocol")

plot3 <- ggplot(data.frame(x = x, coverage = s1r1_sorted), aes(x = x)) +
  geom_line(aes(y = coverage), color= "#dd5182", size = 0.8) +
    geom_segment(aes(x = 0, y = 0.5, xend =  14.5, yend = 0.5),
               linetype = "dashed", 
               color = "gray")+
  geom_segment(aes(x = 14.5, y = 0, xend =  14.5, yend = 0.5),
               linetype = "dashed", 
               color = "gray")+
  xlab("Gene body percentile (5'->3')") +
  ylab("Relative coverage") +
  theme_classic() +
  ggtitle("Live-dissociated libraries")

p<-grid.arrange(plot1, plot2, plot3, ncol = 3)
#save
ggsave( "/drives/raid/AboobakerLab/kriszta/ACME_vs_live/coverage.png",plot=p,  height=3, width=9)
```



```{r}
#correlation plots between replicates of new data
log_DE_matrix<- log(DE_matrix_all+1)
log_DE_matrix
Kriszta_G2<- ggplot(log_DE_matrix, aes(x = Kriszta_G2_1, y =Kriszta_G2_2))+ 
  geom_point(color="#444e86", size=0.3)+ 
  ylab("2nd replicate ln(TPM)")+ xlab("1st replicate ln(TPM)")+ 
  ggtitle("Expression correlation \nbetween G2 replicates")+
  stat_cor(method="pearson")+theme_classic(base_size=10)

Kriszta_G1<- ggplot(log_DE_matrix, aes(x = Kriszta_G1_1, y =Kriszta_G1_2))+ 
  geom_point(color="#a5aed5",size=0.3)+  
  ylab("2nd replicate ln(TPM)")+ xlab("1st replicate ln(TPM)")+ 
  ggtitle("Expression correlation \nbetween G1 replicates")+
  stat_cor(method="pearson")+theme_classic(base_size=10)


correlation_plots<- ggarrange(Kriszta_G2, Kriszta_G1)
correlation_plots
ggsave("/drives/raid/AboobakerLab/kriszta/ACME_vs_live/Figures/correlation_plots.png", plot=correlation_plots, width=5.5, height=3)

```



```{r}
#average tpm values across replicates for each sample type
DE_matrix$live_X1_avg<- (DE_matrix$Aboobaker_X1_1+ DE_matrix$Aboobaker_X1_2+ DE_matrix$Pearson_X1_1+ DE_matrix$Pearson_X1_2+ DE_matrix$Reddien_X1_1+ DE_matrix$Reddien_X1_2+  DE_matrix$Sanchez_X1_1+ DE_matrix$Sanchez_X1_2+ DE_matrix$Sanchez_X1_3+ DE_matrix$Sanchez_X1_4)/10

DE_matrix$live_nonX1<- ((DE_matrix$Pearson_X2_1+ DE_matrix$Pearson_X2_2+ DE_matrix$Reddien_X2_1+ DE_matrix$Reddien_X2_2)/4)+ ((DE_matrix$Pearson_Xins_1+ DE_matrix$Pearson_Xins_2+ DE_matrix$Sanchez_Xins_1+ DE_matrix$Sanchez_Xins_2+ DE_matrix$Sanchez_Xins_3+ DE_matrix$Sanchez_Xins_4)/6) /2
DE_matrix$ACME_G2_avg<- (DE_matrix$Aboobaker_G2_3+ DE_matrix$Aboobaker_G2_4+DE_matrix$Kriszta_G2_1+DE_matrix$Kriszta_G2_2)/4
DE_matrix$ACME_G1_avg<- (DE_matrix$Aboobaker_G1_1+ DE_matrix$Aboobaker_G1_2+ DE_matrix$Kriszta_G1_1+DE_matrix$Kriszta_G1_2)/4

#store the averages in a new data frame
averaged_matrix <- DE_matrix[, 29:32]

```


## 3. PCA

```{r}
#CREATE PCA TABLE
pca_table_DE<-as.data.frame(t(DE_matrix_pca))


pca_table_DE$library<- row.names(pca_table_DE)
pca_table_DE<-
  select(pca_table_DE, library, everything())
pca_table_DE$library <- sub("_[^_]*$", "", pca_table_DE$library)

pca_table_DE<- as.data.frame(pca_table_DE)


pca_table_DE$library

pca_DE <- prcomp(pca_table_DE[,c(2:ncol(pca_table_DE))],
                   center = TRUE)
summary(pca_DE)
str(pca_DE)


```


```{r}
#plot PCA table
pca.plot_DE <- ggplot2::autoplot(pca_DE, x = 1, 
                  y = 2,
                          data = pca_table_DE, color="library")
pca.plot_DE

```

```{r}
#prepare annotations 

pca_table_DE$lab<- c("Aboobaker","Aboobaker","Aboobaker","Aboobaker","Aboobaker","Aboobaker",  "Kriszta", "Kriszta", "Kriszta", "Kriszta", "Pearson", "Pearson","Pearson","Pearson","Reddien", "Reddien", "Reddien", "Reddien", "Sanchez", "Sanchez","Sanchez","Sanchez","Sanchez","Sanchez","Sanchez","Sanchez")

pca_table_DE$cycle<- c("Neoblasts",
                       "Neoblasts", 
                       "Neoblasts",
                       "Neoblasts", 
                       "Non-neoblasts", 
                       "Non-neoblasts", 
                       "Non-neoblasts", 
                       "Neoblasts", 
                         "Non-neoblasts", 
                        "Neoblasts", 
                       "Neoblasts",
                       "Neoblasts", 
                       "Non-neoblasts" , 
                       "Non-neoblasts",
                       "Neoblasts",
                       "Neoblasts",
                       "Non-neoblasts" , 
                       "Non-neoblasts",
                       "Neoblasts",
                       "Neoblasts", 
                       "Neoblasts", 
                       "Neoblasts", 
                       "Non-neoblasts", 
                       "Non-neoblasts", 
                       "Non-neoblasts", 
                       "Non-neoblasts")

pca_table_DE$dissociation<- c("Live", "Live", 
                              "Fixed", "Fixed", "Fixed", "Fixed", 
                              "Fixed", "Fixed", "Fixed", "Fixed", "Live", 
                              "Live", "Live","Live", "Live",  "Live", "Live", "Live", "Live", "Live", "Live", "Live", "Live", "Live", "Live", "Live")

```

```{r}
#plot the PCA plot; group libraries based on dissociation method

pca.plot_DE_dissociation <- ggplot2::autoplot(pca_DE, x = 1, 
                  y = 2,
                          data = pca_table_DE, color="dissociation", size=3 )+ 
  scale_color_manual(values=c("#444e86", "#dd5182")) +
   labs(
         colour = "Dissociation \nmethod:"
        )+
 theme_classic()+ theme(
  legend.title = element_text(size = 18),
  legend.text= element_text(size = 18), 
  legend.position="bottom"
)

pca.plot_DE_dissociation

#save the plot
ggsave("/drives/raid/AboobakerLab/kriszta/ACME_vs_live/Figures/pca.plot_dissociation.png",plot= pca.plot_DE_dissociation, height=6.5, width=6)


```

```{r}
#plot the PCA plot; group libraries based on cell cycle stage
pca.plot_DE_cycle <- ggplot2::autoplot(pca_DE, x = 1, 
                  y = 2,
                          data = pca_table_DE, color="cycle", size=3 )+ 
  scale_color_manual(values=c("#A8CD9F", "#FBA834")) +
   labs(
         colour = "Library \nenriched in:"
        )+
 theme_classic()+ theme(
  axis.title = element_text(size = 18),
  legend.title = element_text(size = 18),
  legend.text= element_text(size = 18), 
    legend.position="bottom"
)

pca.plot_DE_cycle


ggsave("/drives/raid/AboobakerLab/kriszta/ACME_vs_live/Figures/pca.plot_DE_cycle.png",plot= pca.plot_DE_cycle, height=6.5, width=6)


```






```{r}
#create scree plot
library(factoextra)
fviz_eig(pca_DE, addlabels = TRUE)

eigenvalues<-as.data.frame(matrix(nrow=5, ncol=0))
eigenvalues$PC_number<- 1:5
eigenvalues$explained_variation<-c(46, 36.7, 5, 4.1, 2.9)
eigenvalues

#make scree plot more aesthetic using ggplot
scree_plot<- ggplot(eigenvalues, aes(x=PC_number, y=explained_variation)) +
  geom_bar(stat = "identity", fill = "#444e86") + # Change fill color here
  labs(x = "Principal Component", y = "Percentage of explained variance") +

  theme_classic()+theme(
  #plot.title = element_text(size = 13),
  axis.title = element_text(size = 18),
  legend.title = element_text(size = 18),
  legend.text= element_text(size = 18)
)



ggsave("/drives/raid/AboobakerLab/kriszta/ACME_vs_live/Figures/scree_plot.png", plot= scree_plot, width=5.5,  height=5)

```
#4 DESEQ2

```{r}
#prepare the datasets for the comparison- only select datasets that contail "G2" and "X1" in their names 
counts<- na.omit(counts)
counts<- counts[rownames(counts) %in% rownames (ratio_high), ]
counts_comp<-  counts[, grepl("G2|X1", colnames(counts))]

#select the metadata based on the same principles 
metadata_comp<- as.data.frame(metadata[grepl("G2|X1", metadata$cells), ])

```

```{r echo=FALSE}
dds_comp<- DESeqDataSetFromMatrix(countData=counts_comp, colData=metadata_comp, design= ~cells)
keep<-rowSums(counts(dds_comp))>=100
dds_comp<-dds_comp[keep, ]
#relevel and select ACME G2 as baseline for comparison
dds_comp$cells<- relevel(dds_comp$cells, ref="G2")
#run deseq
deseq_comp<- DESeq(dds_comp)

```

```{r}

res_comp<- results(deseq_comp, alpha=0.05)
#summary of results
head(res_comp)

```


```{r include=FALSE}
# order by padj
res_comp <- res_comp[order(res_comp$padj),] 

#create a table containing the most important differential expression data 
results_comp <- data.frame(
  genes = rownames(res_comp),
  log2FoldChange = res_comp$log2FoldChange,
  padj = res_comp$padj,
  lfcSE = res_comp$lfcSE,
  stat = res_comp$stat,
  baseMean = res_comp$baseMean,
  up_down = ifelse(res_comp$log2FoldChange > 1, "Up",
                   ifelse(res_comp$log2FoldChange < -1, "Down", "None")),
  significant = ifelse(res_comp$padj < 0.05, "Sig", "Not_Sig")
)

volcano_comp<-results_comp


```


```{r}

volcano_comp<- volcano_comp[!is.na(volcano_comp$padj),]
#create a direct correspondence with the Rink annotation to be able to browse on Planmine 
matches_volcano <- match(volcano_comp$genes, gene_correspondence$V4)
volcano_comp$GeneID <- gene_correspondence$V1[matches_volcano]
write.csv(volcano_comp, file="/drives/raid/AboobakerLab/kriszta/ACME_vs_live/volcanodata.csv")



#create a volcano object and plot it 
volcano_object_comp<-volcanor(volcano_comp, p= "padj", effect_size="log2FoldChange", snp="genes")
comp_volcano_plot_interactive<- volcanoly(volcano_object_comp, effect_size_line= c(-2, 2), genomewideline = -log10(1e-05), col="a9a9a9", highlight_color = "#367DA2", title="Differentially enriched transcripts in X1 cells(r) VS G2(l)")
comp_volcano_plot_interactive

#define significantly misregulated transcripts 
subset_volcano_comp_G2<- volcano_comp[(volcano_comp$log2FoldChange < -1 ) & volcano_comp$padj < 0.05, ]
subset_volcano_comp_X1<- volcano_comp[(volcano_comp$log2FoldChange > 1) & volcano_comp$padj <0.05, ]




matches <- match(subset_volcano_comp_G2$genes, gene_correspondence$V4)
subset_volcano_comp_G2$GeneID <- gene_correspondence$V5[matches]

matches_live <- match(subset_volcano_comp_X1$genes, gene_correspondence$V4)
subset_volcano_comp_X1$GeneID <- gene_correspondence$V5[matches_live]

```

```{r}
#manually annotated the volcano file 
volcano_comp<- read.csv("/drives/raid/AboobakerLab/kriszta/ACME_vs_live/annotations_functional/final_volcanodata_annotated.csv")

#plot 
volcanoplot<-ggplot(data = volcano_comp, aes(x = log2FoldChange, y = -log10(padj), col = up_down, label = Annotation
                                )) +
  geom_vline(xintercept = c(-1, 1), col = "gray", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "gray", linetype = 'dashed') + 
  geom_point(size = 0.5) + 
  scale_color_manual(values = c("#444e86", "grey",  "#dd5182"))+ 
  coord_cartesian(ylim = c(0, 100), xlim = c(-10, 10)) + 
  scale_x_continuous(breaks = seq(-10, 10, 2)) + 
 labs(color = NULL) +ggtitle("Differentially expressed genes between fixed and live-dissociated neoblasts")+
   theme_classic()+ 
geom_text_repel(size=3, segment.size = 0.2, min.segment.length = 0.5, box.padding = unit(0.2, "lines"), force = 10)
volcanoplot
ggsave("/drives/raid/AboobakerLab/kriszta/ACME_vs_live/Figures/volcanoplot.png", plot=volcanoplot, width=9, height=6)
```


```{r}


#filter transcription factors 
tf_volcano<- volcano_comp[volcano_comp$genes %in% tf$Neiro, ]
tf_volcano <- merge(tf_volcano, tf[, c("Neiro", "Symbol")], by.x = "genes", by.y = "Neiro", all.x = TRUE)
 for (i in 1:nrow(tf_volcano)){
if(tf_volcano$up_down[i]=="None"){
   tf_volcano$Symbol[i]<-NA
}}
tf_volcano

#make volcano plot with transcription factors 
ggplot(data = tf_volcano, aes(x = log2FoldChange, y = -log10(padj), col = up_down, label = tf
                                )) +
  geom_vline(xintercept = c(-1, 1), col = "gray", linetype = 'dashed') +
  geom_hline(yintercept = -log10(0.05), col = "gray", linetype = 'dashed') + 
  geom_point(size = 0.5) + 
  scale_color_manual(values = c("#444e86" , "grey", "#dd5182"))+ 
  coord_cartesian(ylim = c(0, 25), xlim = c(-5, 5)) + # since some genes can have minuslog10padj of inf, we set these limits

  scale_x_continuous(breaks = seq(-10, 10, 2)) + 
 labs(color = NULL) +
   theme_classic()+  
geom_text_repel()
ggsave("/drives/raid/AboobakerLab/kriszta/ACME_vs_live/Figures/tf_volcano.png", width=9, height=6)


tf_volcano_G2<- tf_volcano[tf_volcano$up_down== "Down" & tf_volcano$significant=="Sig",]
write.csv(tf_volcano_G2, file="/drives/raid/AboobakerLab/kriszta/ACME_vs_live/annotations_functional/transcription_factors_G2.csv")

tf_volcano_X1<- tf_volcano[tf_volcano$up_down== "Up" & tf_volcano$significant=="Sig",]
write.csv(tf_volcano_X1, file="/drives/raid/AboobakerLab/kriszta/ACME_vs_live/annotations_functional/transcription_factors_X1.csv")

```


# 5. GO 

```{r}
#make lists of upregulated and downregulated genes
myInterestingGenes_up_in_G2comp <- subset_volcano_comp_G2$genes 

geneList_up_in_G2comp <- factor(as.integer(geneNames %in% myInterestingGenes_up_in_G2comp))
names(geneList_up_in_G2comp) <- geneNames
str(geneList_up_in_G2comp)

 
myInterestingGenes_up_in_livecomp <- subset_volcano_comp_X1$genes 
geneList_up_in_livecomp <- factor(as.integer(geneNames %in% myInterestingGenes_up_in_livecomp))
names(geneList_up_in_livecomp) <- geneNames
str(geneList_up_in_livecomp)


```

```{r}
#BIOLOGICAL PROCESS- ACME data
GO_up_G2 <- new("topGOdata", ontology = "BP", allGenes = geneList_up_in_G2comp,
              annot = annFUN.gene2GO, gene2GO = filtered_geneID2GO, nodeSize = 10)
resultFis_up_G2 <- runTest(GO_up_G2, algorithm = "classic", statistic = "fisher") 
Res_up_G2 <- GenTable(GO_up_G2, classicFisher = resultFis_up_G2, ranksOf = "classicFisher", topNodes = length(resultFis_up_G2@score))
Res_up_G2$classicFisher<- as.numeric(Res_up_G2$classicFisher)

#plot top 20 most significantly enriched GO terms in ACME data
Res_up_g2<-ggplot(filter(Res_up_G2[1:20,], classicFisher < 0.05), aes(x =Significant/Expected, y=reorder(Term, classicFisher), fill = classicFisher)) + geom_bar(stat = "identity") +
theme(axis.title = element_blank(), axis.text = element_text(size = 12))+ theme_classic()

```

```{r}
#make plot more aesthetic 
mypalette <- brewer.pal(3, "YlOrRd")
ggplot(filter(Res_up_G2[1:20,], classicFisher < 0.05)) +
  geom_point(aes(x = classicFisher, 
                 y = reorder(Term, classicFisher), 
                 color = Significant), 
             size = 2) + 
  theme_bw() +
  theme(axis.text.x = element_text(size = rel(1.15)),
        axis.title = element_text(size = rel(1.15)),
        plot.title = element_text(hjust = 0.5, face = "bold"),
        legend.title = element_text(size = rel(1.15), hjust = 0.5, face = "bold")) +
  xlim(0,0.05)+
  xlab("P value") +
  ylab("Top 20 significant GO terms") + 
  ggtitle("Top 20 GO terms significantly enriched in fixed data") +  
  scale_color_gradientn(name = "Number of \nassociated genes", colors = mypalette)

```





```{r}
#cluster GO terms based on semantic similarty
go_analysis <- subset(Res_up_G2, classicFisher < 0.05)
simMatrix <- calculateSimMatrix(go_analysis$GO.ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")

scores <- setNames(-log10(go_analysis$classicFisher), go_analysis$GO.ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.85,
                            orgdb="org.Hs.eg.db")

options(ggrepel.max.overlaps = Inf)
scatterPlot(simMatrix, reducedTerms, labelSize = 5)
treemapPlot(reducedTerms)
ggsave("GO_fixed.png")
```

```{r}
#repeat for live data 
GO_up_X1 <- new("topGOdata", ontology = "BP", allGenes = geneList_up_in_livecomp,
              annot = annFUN.gene2GO, gene2GO = filtered_geneID2GO, nodeSize = 10) 
resultFis_up_X1 <- runTest(GO_up_X1, algorithm = "classic", statistic = "fisher") 
Res_up_X1 <- GenTable(GO_up_X1, classicFisher = resultFis_up_X1, ranksOf = "classicFisher", topNodes = length(resultFis_up_X1@score))
Res_up_X1$classicFisher<- as.numeric(Res_up_X1$classicFisher)
Res_up_X1
```

```{r}
go_analysis <- subset(Res_up_X1, classicFisher < 0.05)
simMatrix <- calculateSimMatrix(go_analysis$GO.ID,
                                orgdb="org.Hs.eg.db",
                                ont="BP",
                                method="Rel")

scores <- setNames(-log10(go_analysis$classicFisher), go_analysis$GO.ID)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.85,
                            orgdb="org.Hs.eg.db")

options(ggrepel.max.overlaps = Inf)
scatterPlot(simMatrix, reducedTerms, labelSize=5)

```

```{r}

mypalette <- brewer.pal(3, "YlOrRd")
ggplot(filter(Res_up_X1[1:60,], classicFisher < 0.05)) +
  geom_point(aes(x = classicFisher, 
                 y = reorder(Term, classicFisher), 
                 color = Significant), 
             size = 2) +  # Moved `size = 2` outside `aes`
  theme_bw() +
  theme(axis.text.x = element_text(size = rel(1.15)),
        axis.title = element_text(size = rel(1.15)),
        plot.title = element_text(hjust = 0.5, face = "bold"),
        legend.title = element_text(size = rel(1.15), hjust = 0.5, face = "bold")) +
  xlim(0,0.05)+
  xlab("P value") +
  ylab("Top 20 significant GO terms") + 
  ggtitle("Top 20 GO terms significantly enriched in fixed data") +  
  scale_color_gradientn(name = "Number of \nassociated genes", colors = mypalette)

```


```{r}
Res_up_X1$classicFisher[1:5]<- 0
Res_up_X1
significant_terms<- Res_up_X1[Res_up_X1$classicFisher<0.05, ]
significant_terms

genes_for_significant_terms <- list()

significant_terms
for (term in significant_terms$GO.ID) {
  genes_for_significant_terms[[term]] <- genesInTerm(GO_up_X1, term)[[1]]
}

names(genes_for_significant_terms) <- significant_terms$GO.ID
significant_terms
```

```{r}

filtered_genes <- volcano_comp[volcano_comp$log2FoldChange > 1 & volcano_comp$padj < 0.05, "genes"]
filtered_genes_for_significant_terms <- list()

# loop through each significant GO term and get genes corresponding to the given GO term
for (term in significant_terms$GO.ID) {
  term_genes <- genesInTerm(GO_up_X1, term)[[1]]
  matching_genes <- intersect(term_genes, filtered_genes)
  if (length(matching_genes) > 0) {
    filtered_genes_for_significant_terms[[term]] <- matching_genes
  }
}


significant_filtered_genes_df <- data.frame(
  GO_ID = rep(names(filtered_genes_for_significant_terms), 
              sapply(filtered_genes_for_significant_terms, length)),
  Gene = unlist(filtered_genes_for_significant_terms)
)

filtered_genes_with_desc <- merge(significant_filtered_genes_df, significant_terms, by.x = "GO_ID", by.y = "GO.ID")

# Display the results
print(filtered_genes_with_desc)
#write.csv(filtered_genes_with_desc, file="/drives/raid/AboobakerLab/kriszta/ACME_vs_live/Figures/X1_GO.csv")

```



```{r}
# Repeat for ACME data
significant_terms<- Res_up_G2[Res_up_G2$classicFisher<0.05, ]

genes_for_significant_terms <- list()

for (term in significant_terms$GO.ID) {
  genes_for_significant_terms[[term]] <- genesInTerm(GO_up_G2, term)[[1]]
}

names(genes_for_significant_terms) <- significant_terms$GO.ID
genes_for_significant_terms

```

```{r}
filtered_genes<- NA

filtered_genes <- volcano_comp[volcano_comp$log2FoldChange <  -1 & volcano_comp$padj < 0.05, "genes"]

filtered_genes_for_significant_terms <- list()

for (term in significant_terms$GO.ID) {

  term_genes <- genesInTerm(GO_up_G2, term)[[1]]

  matching_genes <- intersect(term_genes, filtered_genes)

  if (length(matching_genes) > 0) {
  
    filtered_genes_for_significant_terms[[term]] <- matching_genes
  }
}



significant_filtered_genes_df <- data.frame(
  GO_ID = rep(names(filtered_genes_for_significant_terms), 
              sapply(filtered_genes_for_significant_terms, length)),
  Gene = unlist(filtered_genes_for_significant_terms)
)

filtered_genes_with_desc <- merge(significant_filtered_genes_df, significant_terms, by.x = "GO_ID", by.y = "GO.ID")
filtered_genes_with_desc

print(filtered_genes_with_desc)
#write.csv(filtered_genes_with_desc, file="/drives/raid/AboobakerLab/kriszta/ACME_vs_live/Figures/G2_GO.csv")

```


