
# Bioinformatic preprocessing 

```{r}
#load required packages 
library(Rsamtools)
library(GenomicFeatures)
library(GenomicAlignments)
library(Category)
library(Rsubread)
library(dplyr)
library(edgeR)
```

## 1. Downloading files from SRA

```{bash}
#use fasterq-dump (https://github.com/ncbi/sra-tools/wiki/HowTo:-fasterq-dump) to download FASTA files from the Sequence Read Archive (SRA)
fasterq-dump SRR2009674 # rename Pearson_X1_1.fastq
fasterq-dump SRR2009675 # rename Pearson_X1_2.fastq
fasterq-dump SRR2009676 # rename Pearson_X2_1.fastq
fasterq-dump SRR2009677 # rename Pearson_X2_2.fastq
fasterq-dump SRR2009678 # rename Pearson_Xins_1.fastq
fasterq-dump SRR2009679 # rename Pearson_Xins_2.fastq
fasterq-dump SRR1302023 # rename Reddien_X1_1.fastq
fasterq-dump SRR1302024 # rename Reddien_X1_2.fastq
fasterq-dump SRR1302025 # rename Reddien_X2_1.fastq
fasterq-dump SRR1302026 # rename Reddien_X2_2.fastq
fasterq-dump SRR2407874 # rename Sanchez_X1_1.fastq
fasterq-dump SRR2407875 # rename Sanchez_X1_2.fastq
fasterq-dump SRR2407876 # rename Sanchez_X1_3.fastq
fasterq-dump SRR2407877 # rename Sanchez_X1_4.fastq
fasterq-dump SRR2407878 # rename Sanchez_Xins_1.fastq
fasterq-dump SRR2407879 # rename Sanchez_Xins_2.fastq
fasterq-dump SRR2407880 # rename Sanchez_Xins_3.fastq
fasterq-dump SRR2407881 # rename Sanchez_Xins_4.fastq

```


## 2. Initial quality control 
```{bash}
#single end 
fastqc Pearson_X1_1.fastq
fastqc Pearson_X1_2.fastq
fastqc Pearson_X2_1.fastq
fastqc Pearson_X2_2.fastq
fastqc Pearson_Xins_1.fastq
fastqc Pearson_Xins_2.fastq
fastqc Reddien_X1_1.fastq
fastqc Reddien_X1_2.fastq
fastqc Reddien_X2_1.fastq
fastqc Reddien_X2_2.fastq
fastqc Sanchez_X1_1.fastq
fastqc Sanchez_X1_2.fastq
fastqc Sanchez_X1_3.fastq
fastqc Sanchez_X1_4.fastq

#paired end (in-house datasets)
fastqc W1_1.fastq #Aboobaker_G2_1 R1
fastqc W1_2.fastq #Aboobaker_G2_1 R2
fastqc W2_1.fastq #Aboobaker_G2_2 R1
fastqc W2_2.fastq #Aboobaker_G2_2 R2

fastqc G1_1_1.fastqc #Aboobaker_G1_1 R1
fastqc G1_1_2.fastqc #Aboobaker_G1_1 R2
fastqc G1_2_1.fastqc #Aboobaker_G1_2 R1
fastqc G1_2_2.fastqc #Aboobaker_G1_2 R2

fastqc Wt1_1.fastqc #Aboobaker_X1_1 R1
fastqc Wt1_2.fastqc #Aboobaker_X1_1 R2
fastqc zero_1.fastqc #Aboobaker_X1_2 R1
fastqc zero_2.fastqc #Aboobaker_X1_2 R2
fastqc R1_G1.fastqc #Aboobaker_X1_2 R2
fastqc R1_G2.fastqc #Aboobaker_X1_2 R2
fastqc R2_G2.fastqc #Aboobaker_X1_2 R2
fastqc R2_G1.fastqc #Aboobaker_X1_2 R2
```

```{bash}
#run multiqc on the directory containing all fastqc reports to make a quality control overview report of all libraries 
multiqc . 
```


## 3. Trimming  and second round of quality control 

Step 1: quality trimming 
  - removes the low-quality base calls from the 3' end of the sequence 
Step 2: adapter trimming 
  - auto-detects adaptors and removes them 
  - beacause ACME libraries contained a Clontech primer used in cDNA synthesis that is not automatically detected, I included them in a     separate file and specified their removal
Step 3: removing short sequences 
  - removes sequences that are cut too short; in the paired end mode, it removes the mate pair of the short sequence too to avoid           alignment problems

```{bash}
#paired-end (for this step, I separated paired-end and single-end libraries into different directories)
#trimmomatic_adaptorrem.sh

in=/drives/raid/AboobakerLab/*name*/ACME_vs_live/ACME_RNA-seq/raw_data #specifies location of raw paried-end fastq files 
    out=/drives/raid/AboobakerLab/*name*/ACME_vs_live/ACME_RNA-seq/bioinformatics/trimmed_data #specifies output location
    adapters=/drives/raid/AboobakerLab/*name*/ACME_vs_live/ACME_RNA-seq/bioinformatics/adapters.fasta #specifies files containing the adaptor

paired_files_1=$(ls -R $in | grep _1.fastq) #grabs names of each sample (without the _1.fastq or _2.fastq at the end )

#for each sample, use the R1 and R2 files to perform trimming 
    for i in $paired_files_1; do
        f=$(basename $i _1.fastq) 
        echo "Running on library $f"

java -jar /drives/raid/AboobakerLab/software/Trimmomatic-0.39/trimmomatic-0.39.jar PE ${in}/${f}_1.fastq ${in}/${f}_2.fastq ${out}/${f}_1_out_paired.fq ${out}/${f}_1_out_unpaired.fq ${out}/${f}_2_out_paired.fq ${out}/${f}_2_out_unpaired.fq -threads 16 ILLUMINACLIP:${adapters}:2:30:10 SLIDINGWINDOW:4:20 LEADING:10 TRAILING:10 MINLEN:36
 
 echo ${out}/${f}_1_out_paired.fq ${out}/${f}_1_out_unpaired.fq ${out}/${f}_2_out_paired.fq ${out}/${f}_2_out_unpaired.fq
 
   done
```

```{bash}
#single-end 
#use this code for each file (could also rewrite as a loop like in the case of paired-end data):
java -jar /drives/raid/AboobakerLab/software/Trimmomatic-0.39/trimmomatic-0.39.jar SE Reddien_X2_2.fastq Reddien_X2_2_out_trimmed.fq -threads 16 ILLUMINACLIP:/drives/raid/AboobakerLab/*name*/ACME_vs_live/preprocessing/ACME_RNA-seq/bioinformatics/adapters.fasta:2:30:10 SLIDINGWINDOW:4:20 LEADING:10 TRAILING:10 MINLEN:36

```

Following trimming, perform a second round of quality control using fastqc and multiqc. 

## 4. Mapping to reference genome 

Index the reference genome:
```{bash}
#create a file to store the index files and run this: 
hisat2-build /drives/ssd1/simon/smed/snp_analysis/haplocall/dd_Smes_g4.fasta dd_Smed_g4_index_new

```


```{bash}
#paired end:
#read_aligner.sh
#nohup ./read_aligner.sh > mapping.out.new 2>&1 &

ls *_1_out_paired.fq |sort -u |while read id;do
  f=$(basename $id _1_out_paired.fq)
  
echo "Running on $f"
#echo $id
hisat2 -p 8 -x /drives/raid/AboobakerLab/*name*/ACME_vs_live/ACME_RNA-seq/bioinformatics/mapped_reads/hisat2_index/dd_Smed_g4_index_new -1 *${f}*_1_out_paired.fq   -2 *${f}*_2_out_paired.fq | samtools view -Sb | samtools sort -o /drives/raid/AboobakerLab/*name*/ACME_vs_live/remaining_analysis/aligned_data/${f}.bam
done 


#echo "export PATH=$PATH:/drives/raid/AboobakerLab/software/samtools-1.17/" >> ~/.bashrc
#source ~/.bashrc

```

```{bash}
#single end:
hisat2 -p 8 -x /drives/raid/AboobakerLab/*name*/ACME_vs_live/ACME_RNA-seq/bioinformatics/mapped_reads/hisat2_index/dd_Smed_g4_index_new -U Reddien_X2_2_out_trimmed.fq | samtools view -Sb | samtools sort -o Reddien_X2_2.bam 
```

### Extract unmapped files and map to the mitochondrial genome

```{bash}

#nohup ./map_to_mitochondria.sh > mapping.out_mitochondria 2>&1 &

samtools view -f 4 -F 8 -b /drives/raid/AboobakerLab/*name*/ACME_vs_live/ACME_RNA-seq/bioinformatics/mapped_reads/W1.bam > W1_1_unmapped.bam #extract mate 1
samtools view -f 4 -f 8 -b /drives/raid/AboobakerLab/*name*/ACME_vs_live/ACME_RNA-seq/bioinformatics/mapped_reads/W1.bam > W1_2_unmapped.bam  #extract mate 2

#significantly more mate 2 unmapped reads -> I am going to map them as non-paired 
bedtools bamtofastq -i unmapped_W1.bam -fq unmapped_W1.fastq
hisat2 -p 8 -x /drives/raid/AboobakerLab/*name*/ACME_vs_live/ACME_RNA-seq/bioinformatics/mapped_reads/index_mitochondria/mitochonrial_index -U unmapped_W1.fasta | samtools view -Sb | samtools sort -o unmapped_W1_output.bam 


```


## 5. Quantifying transcript abundance 

```{r}

#paired unstranded

exonCounts_pu <- featureCounts(readLines("/drives/raid/AboobakerLab/*name*/ACME_vs_live/library_all.txt")[1:10],
                                               annot.ext =  "/drives/raid/AboobakerLab/*name*/ACME_vs_live/annotations_functional/Smed.transcripts.fasta.transdecoder.genome.gtf",
                                               isGTFAnnotationFile= TRUE,
                                               GTF.featureType = "exon",
                                               GTF.attrType = "gene_id",
                                               useMetaFeatures= TRUE,
                                               allowMultiOverlap = FALSE,
                                               minOverlap = 2,
                                               strandSpecific = 0, 
                                               isPairedEnd= TRUE,
                                               nthreads = 8)

```

```{r}
#unpaired unstranded
exonCounts_uu <- featureCounts(readLines("/drives/raid/AboobakerLab/*name*/ACME_vs_live/library_all.txt")[11:20],
                annot.ext = "/drives/raid/AboobakerLab/*name*/ACME_vs_live/annotations_functional/Smed.transcripts.fasta.transdecoder.genome.gtf",
                isGTFAnnotationFile= TRUE,
                GTF.featureType = "exon",
                GTF.attrType = "gene_id",
                useMetaFeatures= TRUE,
                allowMultiOverlap = FALSE,
                minOverlap = 2,
                strandSpecific = 0, #0 is non-stranded; 1 is stranded; 2 is reversely stranded
                isPairedEnd= FALSE,
                nthreads = 8)
```

```{r}
#unpaired reverse stranded
exonCounts_ur <- featureCounts(readLines("/drives/raid/AboobakerLab/*name*/ACME_vs_live/library_all.txt")[21:28],
                annot.ext = "/drives/raid/AboobakerLab/*name*/ACME_vs_live/annotations_functional/Smed.transcripts.fasta.transdecoder.genome.gtf",
                isGTFAnnotationFile= TRUE,
                GTF.featureType = "exon",
                GTF.attrType = "gene_id",
                useMetaFeatures= TRUE,
                allowMultiOverlap = FALSE,
                minOverlap = 2,
                strandSpecific = 2, #0 is unstranded; 1 is stranded; 2 is reversely stranded
                isPairedEnd= FALSE,
                nthreads = 8)
```

```{r}
#create count matrices
merged_counts_temp<- merge(exonCounts_pu$counts, exonCounts_uu$counts, by="row.names", all=TRUE)
rownames(merged_counts_temp)<- merged_counts_temp$Row.names
merged_counts_temp<- merged_counts_temp[,-1]

merged_counts_tosave<- merge(merged_counts_temp, exonCounts_ur$counts, by="row.names", all=TRUE)
rownames(merged_counts_tosave)<- merged_counts_tosave$Row.names
merged_counts_tosave<- merged_counts_tosave[,-1]

merged_counts_tosave<- merged_counts_tosave[, -c(3, 4, 9, 10)]
#reformat column names
colnames(merged_counts_tosave)<- c("Aboobaker_X1_1", "Aboobaker_X1_2", "Aboobaker_G2_1", "Aboobaker_G2_2", "Aboobaker_G1_1", "Aboobaker_G1_2", "Pearson_X1_1", "Pearson_X1_2", "Pearson_X2_1", "Pearson_X2_2", "Pearson_Xins_1", "Pearson_Xins_2", "Reddien_X1_1", "Reddien_X1_2", "Reddien_X2_1", "Reddien_X2_2", "Sanchez_X1_1", "Sanchez_X1_2", "Sanchez_X1_3", "Sanchez_X1_4", "Sanchez_Xins_1", "Sanchez_Xins_2", "Sanchez_Xins_3", "Sanchez_Xins_4")

write.csv(merged_counts_tosave, file="/drives/raid/AboobakerLab/*name*/ACME_vs_live/counts_tpm_matrices/counts_all_datasets.csv", row.names=TRUE)

matrix_all_datasets<- data.frame(matrix(ncol = 0, nrow = 24))

matrix_all_datasets$library<- c("Aboobaker_X1_1", "Aboobaker_X1_2", "Aboobaker_G2_1", "Aboobaker_G2_2", "Aboobaker_G1_1", "Aboobaker_G1_2", "Pearson_X1_1", "Pearson_X1_2", "Pearson_X2_1", "Pearson_X2_2", "Pearson_Xins_1", "Pearson_Xins_2", "Reddien_X1_1", "Reddien_X1_2", "Reddien_X2_1", "Reddien_X2_2", "Sanchez_X1_1", "Sanchez_X1_2", "Sanchez_X1_3", "Sanchez_X1_4", "Sanchez_Xins_1", "Sanchez_Xins_2", "Sanchez_Xins_3", "Sanchez_Xins_4")

matrix_all_datasets$dissociation_method<- c("live", "live", "ACME", "ACME", "ACME", "ACME", "live", "live", "live", "live", "live", "live", "live", "live", "live", "live", "live", "live", "live", "live", "live", "live", "live", "live")

matrix_all_datasets$cells<- c("X1", "X1", "G2", "G2", "G1", "G1", "X1", "X1", "X2", "X2", "Xins", "Xins","X1", "X1", "X2", "X2",  "X1", "X1", "X1", "X1", "Xins", "Xins", "Xins", "Xins")


write.csv(matrix_all_datasets, file="/drives/raid/AboobakerLab/*name*/ACME_vs_live/annotations_functional/matrix_all_datasets.csv", row.names=TRUE)
```


## 6. Calculating TPM values 

```{r}

#calculate tpms from featureCounts output
standardised_by_length <- function(counts, lengths) {
  rate <- counts / lengths
  } #gives us reads per kilobase 


tpm_pu <- as.data.frame(standardised_by_length(exonCounts_pu$counts, exonCounts_pu$annotation$Length))

libsum<- c(0)
for (i in 1:ncol(tpm_pu)){
  libsum[i]<-as.numeric(sum(tpm_pu[, i])/1000000)
} #TPM scaling factor

for (i in 1:ncol(tpm_pu)) {
  tpm_pu[, i] <- tpm_pu[, i] / libsum[i]
}

```

```{r}
#calculate tpm values for the live X2 and Xins data
tpm_uu <- as.data.frame(standardised_by_length(exonCounts_uu$counts, exonCounts_uu$annotation$Length))

libsum<- c(0)
for (i in 1:ncol(tpm_uu)){
  libsum[i]<-as.numeric(sum(tpm_uu[, i])/1000000)
} #TPM scaling factor

for (i in 1:ncol(tpm_uu)) {
  tpm_uu[, i] <- tpm_uu[, i] / libsum[i]
}

```

```{r}
#calculate tpm values for the ACME data
tpm_ur <- as.data.frame(standardised_by_length(exonCounts_ur$counts, exonCounts_ur$annotation$Length))

libsum<- c(0)
for (i in 1:ncol(tpm_ur)){
  libsum[i]<-as.numeric(sum(tpm_ur[, i])/1000000)
} #TPM scaling factor

for (i in 1:ncol(tpm_ur)) {
  tpm_ur[, i] <- tpm_ur[, i] / libsum[i]
}

```

```{r}

#merge the first two matrices
merged_tpms<-merge(tpm_pu, tpm_uu, by = "row.names", all = TRUE)
rownames(merged_tpms)<- merged_tpms$Row.names
merged_tpms<- merged_tpms[, -c(1)]

#merge the last matrix too
DE_matrix_all<- merge(merged_tpms, tpm_ur, by = "row.names", all = TRUE)
rownames(DE_matrix_all)<- DE_matrix_all$Row.names
DE_matrix_all<- DE_matrix_all[, -c(1)]

write.csv(DE_matrix_all, file="/drives/raid/AboobakerLab/*name*/ACME_vs_live/counts_tpm_matrices/tpm_all_datasets", row.names=TRUE )

DE_matrix_read<- read.csv ("/drives/raid/AboobakerLab/*name*/ACME_vs_live/counts_tpm_matrices/tpm_all_datasets")
rownames(DE_matrix_read)<- DE_matrix_read$X
DE_matrix_read<- DE_matrix_read[, -1]

DE_matrix<- DE_matrix_read[, -c(3, 4, 9, 10)]
#reformat column names
colnames(DE_matrix)<- c("Aboobaker_X1_1", "Aboobaker_X1_2", "Aboobaker_G2_1", "Aboobaker_G2_2", "Aboobaker_G1_1", "Aboobaker_G1_2", "Pearson_X1_1", "Pearson_X1_2", "Pearson_X2_1", "Pearson_X2_2", "Pearson_Xins_1", "Pearson_Xins_2", "Reddien_X1_1", "Reddien_X1_2", "Reddien_X2_1", "Reddien_X2_2", "Sanchez_X1_1", "Sanchez_X1_2", "Sanchez_X1_3", "Sanchez_X1_4", "Sanchez_Xins_1", "Sanchez_Xins_2", "Sanchez_Xins_3", "Sanchez_Xins_4")

write.csv(DE_matrix, file="/drives/raid/AboobakerLab/*name*/ACME_vs_live/counts_tpm_matrices/tpm_all.csv", row.names=TRUE )

```




